docs/en/developer/create-web-component/VUE.md

```diff
@@ -1,137 +1,445 @@
-# replace {component-name} with the desired name for the component
-vue create {component-name} --default
+Copy **build.gradle** file from @uportal directory of **uPortal-web-components**
+project (or use one from [the appendix](#buildgradle-files)).
 
-# navigate into newly created component folder
-cd {component-name}
+https://github.com/uPortal-contrib/uPortal-web-components/blob/master/%40uportal/build.gradle
 
-# add additional dependencies for legacy browser support
-npm install --save-dev @babel/{cli,plugin-transform-runtime,preset-env}
+If using the one from uPortal-web-components, remove the subprojects line and
+its enclosing brackets; for example:
+
+``` diff
+- subprojects {
+
+    apply plugin: 'java'
+    apply plugin: 'maven'
+
+    def jsonFile = file("${projectDir}/package.json")
+    def parsedJson = new groovy.json.JsonSlurper().parseText(jsonFile.text)
+    project.version = parsedJson.version + '-SNAPSHOT'
+
+    jar {
+        baseName "uportal__${project.name}"
+        from '.'
+        exclude 'build'
+        exclude 'node_modules'
+        into "META-INF/resources/webjars/uportal__${project.name}/${project.version}"
+    }
+
+- }
+```
+
+It should look something like this:
+
+``` gradle
+    apply plugin: 'java'
+    apply plugin: 'maven'
+
+    def jsonFile = file("${projectDir}/package.json")
+    def parsedJson = new groovy.json.JsonSlurper().parseText(jsonFile.text)
+    project.version = parsedJson.version + '-SNAPSHOT'
+
+    jar {
+        baseName "uportal__${project.name}"
+        from '.'
+        exclude 'build'
+        exclude 'node_modules'
+        into "META-INF/resources/webjars/uportal__${project.name}/${project.version}"
+    }
+```
+
+**Note:** to produce the most compact .jar possible to reduce traffic on the
+resource server and speed up browser response, or if you are building on
+Windows, try this [build.gradle for Linux and Mac OS](#buildgradle-for-linux-and-mac-os)
+or this [build.gradle for Windows](#buildgradle-for-windows).
+
+### Add Gradle wrapper to project
+
+Run this:
+
+```
+gradle wrapper --gradle-version=5.1.1
+```
+
+Note: now that the project has a build.gradle file and a Gradle wrapper, an
+IDE like IntelliJ IDEA will recognize it as a Gradle project. You could do
+the rest of the required editing in the IDE.
+
+### Rename HelloWorld.vue
+
+Rename the generated HelloWorld.vue file, replace **{component-name}** with yours:
+
+```
+FROM:
+src/components/HelloWorld.vue
+
+TO:
+src/components/{component-name}.vue
+```
+
+### Edit App.vue
+
+Rename the imports.generated HelloWorld.vue file, replace **{component-name}** with yours:
+
+``` javascript
+// FROM:
+import HelloWorld from './components/HelloWorld.vue'
+
+// TO:
+import HelloWorld from './components/{component-name}.vue'
 ```
 
-### Edit _package.json_
+### Edit package.json
+
+The `{component-name}` after `--name` _**must**_ have a
+hyphen, for example `--name weather-thingy`. Change the following:
+
```

Comment:
🙂 perhaps adding something like "... a hyphen, because it is in the _custom-element_ specification ( ref: [custom-element on MDN](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_custom_elements#High-level_view) ), for example...."

Reply:
Thanks